### PR TITLE
chore: align current_miniblock, current_batch, current_timestamp to point to the latest entities

### LINKED ISSUE #131: Rename current_miniblock
#### 📝 Description

We have 3 similar sounding fields (current_batch, current_timestamp, current_miniblock) - but they have different behavior - the next transaction will be in 'current_batch' BATCH, with current_timestamp timestamp, but with current_miniblock+1 block.

#### 🤔 Rationale

To make it less confusing for people who read the code.

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* Aligns current_miniblock, current_batch, current_timestamp to point to the latest entities. Previously current_miniblock would contain the latest block but others would contain values for the next block
* Fix `compute_hash` type for block_number being `u32` instead of `u64`
* Fixes behavior mining blocks according to the `current_timestamp` rules.

# Why :hand:
* The code should not be confusing to any new contributors and must be consistent.

# Evidence :camera:
* All tests pass
* New tests for creating empty blocks
image

# Notes :memo:
* Fixes #131

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -717,11 +730,14 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
             let mut block_hashes = HashMap::<u64, H256>::new();
             block_hashes.insert(0, H256::zero());
             let mut blocks = HashMap::<H256, Block<TransactionVariant>>::new();
-            blocks.insert(H256::zero(), create_empty_block(0, 0, 1));
+            blocks.insert(
+                H256::zero(),
+                create_empty_block(0, NON_FORK_FIRST_BLOCK_TIMESTAMP, 0),
```
Comment (nbaztec) [inline src/node.rs:735]: Updated the timestamp & batch values here to be consistent with the current values

File: src/utils.rs [hunk 2]
```diff
@@ -98,15 +98,20 @@ pub fn mine_empty_blocks<S: std::fmt::Debug + ForkSource>(
...
-        let (keys, bytecodes) = {
+        let (keys, bytecodes, next_block) = {
             let storage = StorageView::new(&node.fork_storage).to_rc_ptr();
 
             // system_contract.contacts_for_l2_call() will give playground contracts
             // we need these to use the unsafeOverrideBlock method in SystemContext.sol
             let bootloader_code = node.system_contracts.contacts_for_l2_call();
-            let batch_env = node.create_l1_batch_env(storage.clone());
+            let (batch_env, mut next_block) = node.create_l1_batch_env(storage.clone());
+            println!("timestamp {}", next_block.timestamp);
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply src/utils.rs]: Oops that wasn't supposed to be there.

File: src/node.rs [hunk 3]
```diff
@@ -238,13 +238,14 @@ pub struct TransactionResult {
 /// Helper struct for InMemoryNode.
 /// S - is the Source of the Fork.
 pub struct InMemoryNodeInner<S> {
-    /// Timestamp, batch number that will be used by the next block.
+    /// The latest timestamp that was already generated.
+    /// Next block will be current_timestamp + 1
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply src/node.rs:242]: Made the changes

**Review by MexicanAce — DISMISSED** [review]
LGTM

**Comment by dutterbutter** [comment]
@nbaztec please address conflicts

**Review by mm-zk — APPROVED** [review]

File: src/node.rs [hunk 4]
```diff
@@ -277,25 +278,35 @@ type L2TxResult = (
...
 
 impl<S: std::fmt::Debug + ForkSource> InMemoryNodeInner<S> {
-    pub fn create_l1_batch_env<ST: ReadStorage>(&self, storage: StoragePtr<ST>) -> L1BatchEnv {
+    pub fn create_l1_batch_env<ST: ReadStorage>(
+        &self,
+        storage: StoragePtr<ST>,
+    ) -> (L1BatchEnv, NextBlock) {
         let last_l2_block = load_last_l2_block(storage);
-        L1BatchEnv {
+        let next_block = NextBlock::from_current(
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply src/node.rs:290]: Yeah naming's been hard here. I tried to create a context difference here where `NextBlock` (open to any alternate name suggestions) keeps track of information/context for creating the next current block - where `current_block` would usually be a `Block` with details filled in. I guess another name could be `NextBlockContext` or simply `BlockContext` let me know what you think. I agree, the name work is a bit eh.

[the remark this answers is not in the document]
Comment (dutterbutter) [reply src/node.rs:290]: +1 for BlockContext

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**